numba/tests/doc_examples/test_examples.py

```diff
@@ -220,6 +354,204 @@ def g(x, y, res):
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2]])
+            np.testing.assert_array_equal(outvals, correct)
+
+    def test_vectorize_dynamic(self):
+        with captured_stdout():
+            # magictoken.ex_vectorize_dynamic.begin
+            from numba import vectorize
+
+            @vectorize
+            def f(x, y):
+                return x * y
+            # magictoken.ex_vectorize_dynamic.end
+
+            # magictoken.ex_vectorize_dynamic_call_one.begin
+            result = f(3,4)
+            # result == 12
+
+            print(f.types)
+            # ['ll->l']
+            # magictoken.ex_vectorize_dynamic_call_one.end
+
+            self.assertEqual(result, 12)
+            if IS_WIN32:
+                correct = ['ll->q']
+            else:
+                correct = ['ll->l']
+            self.assertEqual(f.types, correct)
+
+            # magictoken.ex_vectorize_dynamic_call_two.begin
+            result = f(1.,2.)
+            # result == 2.0
+
+            print(f.types)
+            # ['ll->l', 'dd->d']
+            # magictoken.ex_vectorize_dynamic_call_two.end
+
+            self.assertEqual(result, 2.0)
+            if IS_WIN32:
+                correct = ['ll->q', 'dd->d']
+            else:
+                correct = ['ll->l', 'dd->d']
+            self.assertEqual(f.types, correct)
+
+            # magictoken.ex_vectorize_dynamic_call_three.begin
+            result = f(1,2.)
+            # result == 2.0
+
+            print(f.types)
+            # ['ll->l', 'dd->d']
+            # magictoken.ex_vectorize_dynamic_call_three.end
+
+            self.assertEqual(result, 2.0)
+            if IS_WIN32:
+                correct = ['ll->q', 'dd->d']
+            else:
+                correct = ['ll->l', 'dd->d']
+            self.assertEqual(f.types, correct)
+
+            # magictoken.ex_vectorize_dynamic_call_four.begin
+            @vectorize
+            def g(a, b):
+                return a / b
+
+            print(g(2.,3.))
+            # 0.66666666666666663
+
+            print(g(2,3))
+            # 0.66666666666666663
+
+            print(g.types)
+            # ['dd->d']
+            # magictoken.ex_vectorize_dynamic_call_four.end
+
+            correct = ['dd->d']
+            self.assertEqual(g.types, correct)
+
+    def test_guvectorize_dynamic(self):
+        with captured_stdout():
+            # magictoken.ex_guvectorize_dynamic.begin
+            from numba import guvectorize
+            import numpy as np
+
+            @guvectorize('(n),()->(n)')
+            def g(x, y, res):
+                for i in range(x.shape[0]):
+                    res[i] = x[i] + y
+            # magictoken.ex_guvectorize_dynamic.end
+
+            # magictoken.ex_guvectorize_dynamic_call_one.begin
+            x = np.arange(5, dtype=np.int64)
+            y = 10
+            res = np.zeros_like(x)
+            # res == array([10, 11, 12, 13, 14])
```

Comment:
I think this comment should be after the call to `g()` - at this point in the code it will be zeroes.